jspoon/src/main/java/pl/droidsonroids/jspoon/ArrayUtils.java

```diff
@@ -0,0 +1,168 @@
+package pl.droidsonroids.jspoon;
+
+class ArrayUtils {
+
+    /**
+     * Converts a given array of wrappers to the same type of primitive array. Null values are
+     * represented as initial value for that primitive type. If the wrapper array is null, will
+     * return null.
+     * @param array wrapper array
+     * @param primitiveType type of the returned primitive array
+     * @throws IllegalArgumentException if the given primitive type is not a class of a primitive
+     * @throws ClassCastException if primitive type does not represent items of the wrapper array
+     */
+    static Object toPrimitive(final Object[] array, Class<?> primitiveType) {
+        if (array == null) {
+            return null;
+        }
+        if (!primitiveType.isPrimitive()) {
+            throw new IllegalArgumentException(primitiveType + " is not a primitive type");
+        }
+        Class<?> pt = primitiveType;
+        if(Boolean.TYPE.equals(pt)) {
+            return toPrimitive((Boolean[]) array);
+        }
+        if(Integer.TYPE.equals(pt)) {
+            return toPrimitive((Integer[]) array);
+        }
+        if(Long.TYPE.equals(pt)) {
+            return toPrimitive((Long[]) array);
+        }
+        if(Short.TYPE.equals(pt)) {
+            return toPrimitive((Short[]) array);
+        }
+        if(Byte.TYPE.equals(pt)) {
+            return toPrimitive((Byte[]) array);
+        }
+        if(Character.TYPE.equals(pt)) {
+            return toPrimitive((Character[]) array);
+        }
+        if(Double.TYPE.equals(pt)) {
+            return toPrimitive((Double[]) array);
+        }
+        if(Float.TYPE.equals(pt)) {
+            return toPrimitive((Float[]) array);
+        }
+        return array;
+    }
+
+    static char[] toPrimitive(Character[] array) {
+        if (array == null) {
+            return null;
+        }
+        final char[] result = new char[array.length];
+        for (int i = 0; i < array.length; i++) {
+            if (array[i] == null) {
+                result[i] = (new char[1])[0];
```

Comment:
I would extract `(new char[1])[0]` to a constant like `DEFAULT_CHAR_VALUE`. Analogously for other primitive types.

Reply:
@koral--  ok, updated to in-line primitive defaults. Such minor detail makes little difference to me, but I would like to get this patched in so I could have the bugs fixed and array support.